core/base/inc/ROOT/StringConv.hxx

```diff
@@ -99,12 +99,18 @@ EFromHumanReadableSize FromHumanReadableSize(std::string_view str, T &value)
       int unit = 1000;
 
       auto result = [coeff,&exp,&unit,&value]() {
-         double v = exp ? coeff * std::pow(unit, exp / 3) : coeff;
-         if (v < std::numeric_limits<T>::max()) {
-            value = (T)v;
-            return EFromHumanReadableSize::kSuccess;
-         } else {
+         auto v = exp ? coeff * std::pow(unit, exp / 3) : coeff;
+         // protection from overflow [-Wimplicit-int-float-conversion]
+         if ((T)v >= std::numeric_limits<T>::max()) {
```

Comment:
This seems wrong.   Shouldn't the cast actually be in the other direction?

Reply:
Printing values in test I see that:
(double)9.22337e+18 -> (long long) -9223372036854775808

I added few protections for this case...

Reply:
But I can simplify code a bit :)

Reply:
> I added few protections for this case, since we are strictly interested in valid > interval of values not including min() and max() values. 

Yes, I see the intent but it seems the code is still incorrect.  For example with 'v' of type int and T==char:
```
root [0] #include <limits>
root [1] ((char)257) >= std::numeric_limits<char>::max()
(bool) false
```
whereas:
```
root [2] 257 >= (int)std::numeric_limits<char>::max()
(bool) true
```

Reply:
hmmm...I see what you mean..